scene/main/viewport.h

```diff
@@ -285,6 +285,11 @@ class Viewport : public Node {
 	Ref<ViewportTexture> default_texture;
 	Set<ViewportTexture *> viewport_textures;
 
+	struct GUITouchState {
+		int index;
```

Comment:
I think TouchFocus is the better name, as you have done with the `Vector` of them: `touch_focuses`.